### PR TITLE
Feature/doc testing

### PR DESCRIPTION
**Pull request by IliassB**
Adding documentation for testing in django-socio-grpc.

Credit to @perissg for the contribution !

### CODE REVIEW

File: docs/testing.rst [hunk 1]
```diff
@@ -4,8 +4,119 @@ Testing
...
+    
+    class TestPost(TestCase):
+        def setup(self):
+            self.fake_grpc = FakeFullAIOGRPC(
+                add_PostControllerServicer_to_server,
+                PostService.as_servicer(),
+            )
+
+        def tearDown(self):
+            self.fake_grpc.close()
```
Comment (legau) [inline docs/testing.rst:57]: I checked and this does nothing, you can remove it

**Review by legau — CHANGES_REQUESTED** [review]
Thanks for the doc, a few comments

File: docs/testing.rst [hunk 2]
```diff
@@ -4,8 +4,119 @@ Testing
...
+
+Now, let's dissect the ``TestPost`` class:
+
+- We recommend using ``TestCase`` from ``django.test`` for testing your service.
+- The ``setup`` method should instantiate a ``FakeFullAIOGRPC``.
+- ``FakeFullAIOGRPC`` is initialized with two parameters: a function to add a servicer to a server, and the servicer itself. In this case, ``add_PostControllerServicer_to_server`` is the function generated by gRPC that adds a ``PostService`` servicer to a server.
+- When you create an instance of ``FakeFullAIOGRPC``, it sets up a mock gRPC server with your servicer. This allows you to make calls to your servicer methods as if they were being handled by a real gRPC server.
+- The close method is used to clean up after your tests are done. It stops the mock gRPC server and releases any resources it was using.
```
Comment (AMontagu) [inline docs/testing.rst:65]: to remove if leni said it's not used

File: docs/testing.rst [hunk 3]
```diff
@@ -4,8 +4,119 @@ Testing
...
+            )
+
+        def tearDown(self):
+            self.fake_grpc.close()
+
+Now, let's dissect the ``TestPost`` class:
+
+- We recommend using ``TestCase`` from ``django.test`` for testing your service.
+- The ``setup`` method should instantiate a ``FakeFullAIOGRPC``.
+- ``FakeFullAIOGRPC`` is initialized with two parameters: a function to add a servicer to a server, and the servicer itself. In this case, ``add_PostControllerServicer_to_server`` is the function generated by gRPC that adds a ``PostService`` servicer to a server.
```
Comment (AMontagu) [inline docs/testing.rst:63]: This is too compact and not enought explicit
make a sub list a two element to explain add_PostControllerServicer_to_server and what is a servicer.
Add a link to the grpc doc: https://grpc.io/docs/languages/python/basics/ saying to look for Starting the server part of the document

File: docs/testing.rst [hunk 4]
```diff
@@ -4,8 +4,119 @@ Testing
 Overview
 --------
 
+The ``django-socio-grpc`` is a powerful library that serves as a layer on top of the Django Rest Framework (DRF). It is designed to enhance the capabilities of Django and DRF, providing additional functionality and features.
+
```
[the remark this answers is not in the document]
Comment (AMontagu) [reply docs/testing.rst:8]: Replace it with the global logic of the mocking class of DSG.
Something like:
- To enable unit test DSG provide class that mock the grpc behavior, replacing the networking call by direct function call, ...

**Review by AMontagu — COMMENTED** [review]
+ miss deprecated statement on FakeGrpc and FakeAIOGrpc

**Pull request merged**